lib/receptor_controller/client/response_worker.rb

```diff
@@ -156,6 +162,27 @@ def check_timeouts(threshold = config.response_timeout)
       logger.error("Exception in maintenance worker: #{err}\n#{err.backtrace.join("\n")}")
     end
 
+    # GZIP recognition
+    # https://tools.ietf.org/html/rfc1952#page-5
+    def gzipped?(data)
+      sign = data.bytes[0..1]
+      return false if sign.size != 2
```

Comment:
I'm not sure we need this guard - it'll just return nil if there isn't anything at index 0 or 1, then the check below will still fail (as we want it to). 
e.g. `>> "s".bytes[0..1][1]   #=> nil`